### DISCUSSION TITLE
Android faill on redirect to localhost/127.0.0.1

### DISCUSSION THREAD

**Discussion by lufiipe** [GodotNuts/GodotFirebase#324] [Q&A] [answered]
Hello
I'm trying to make google authentication. I use Godot version 3.5.
The app works very well in Desktop mode, but unfortunately not with Android.
I have enabled the Permissions "Access Network State" and "Internet" in Android export template.

The redirection to the google authentication page works, but when I validate my credentials, the redirection to http://localhost:8060/ is blocked on my device.

Thanks in advance.

![Screenshot_20221202-181835_Chrome](https://user-images.githubusercontent.com/7064120/205450272-ea2f4507-335b-4ff4-af4f-04de757620af.jpg)

**Comment by fenix-hub** [comment, marked as answer]
Hi @elfilipe ,
Local OAuth2 is not currently supported on Android.
The reason why it works on desktop is because a desktop application is able to work as a webserver and serve localhost to get requests for redirection. On Android that can't be done in this way.
A possible solution is to configure Android deep linking for your Android App, and use that link as a redirection URI for OAuth2. We don't have any working example yet, but simply because nobody asked this feature.
 It is possible anyway, but requries you to use integrate additional modules, like this one https://github.com/timoschwarzer/godot-applinks to get Deep Linking working on Godot

**Comment by lufiipe** [comment]
Thank you for your answer, i'll take a look at the godot-applinks solution.

**Reply by brahma0545** [reply]
@elfilipe I am having the same issue, when I build an app to my Android device, I get the same error, I understand the reason for it but can't fix it, if you have found the solution, can you help me with the code? or the process of how to fix this.
